### ISSUE TITLE
Strange output of SpectrumCQ 

### ISSUE THREAD

**Issue by leonid-s-usov** [MTG/essentia#894]
Does this kind of output from SpecturmCQ algo look valid? I'd expect to see something more closely resembling the usual Spectrum but with logarithmic frequency scale.

The sampling rate is 44100, and it's correctly applied at all levels - the pitch detection works fine (the red number 319 top right, with .99 confidence)

```
    int framesize = 16*1024; //required for constantQ
    int hopsize = 2*1024;
    int sr = samplingRate;
    
    Algorithm* frameCutter  = factory.create("FrameCutter",
                                             "frameSize", framesize,
                                             "hopSize", hopsize,
                                             "silentFrames", "noise"
                                             //, "startFromZero", true,
                                             //, "validFrameThresholdRatio", 1.
                                             );
    Algorithm* window       = factory.create("Windowing",
                                             "type", "hamming",
                                             "size", framesize);
    Algorithm* spec  = factory.create("SpectrumCQ"
                                     ,"sampleRate", sr
                                      , "minFrequency", 100
                                      , "maxFrequency", 2000
                                      , "binsPerOctave", 24
                                      , "threshold", 0.0005
                                      );
    
    Algorithm* hp = factory.create("HighPass", "sampleRate", sr, "cutoffFrequency", 50);
    
    audio->output("signal")                 >>  hp->input("signal");
    hp->output("signal")                 >>  frameCutter->input("signal");
    frameCutter->output("frame")            >>  window->input("frame");
    window->output("frame") >> spec->input("frame");
```

![out](https://user-images.githubusercontent.com/3144935/61993380-cbe4ae00-b073-11e9-92d4-b210a60a5b77.gif)

**Comment by dbogdanov** [comment]
Hi @leonid-s-usov. As have recently found, it's not valid. We've been just refactoring it in the branch [constantq_refactor](https://github.com/MTG/essentia/tree/constantq_refactor). We still need to validate it all works properly.

If you need CQ spectrograms, you can use NSGConstantQ instead: https://mtg.github.io/essentia-labs//news/2019/02/07/invertible-constant-q/

**Comment by leonid-s-usov** [comment]
@dbogdanov how do I use the output of the transform, given it produces a vector of vectors out of a single frame?

**Comment by pabloEntropia** [comment]
@leonid-s-usov NSGConstantQ produces a vector of vectors because it is a time to time/frequency transform. For a single audio frame, it produces a Constant-Q spectrogram of `N` frequency bins by `M` timestamps where `N` depends on the configuration parameters and `M` is the minimum amount of timestamps required for perfect reconstruction.
Use this output considering that, if the input audio frames have any overlapping, consecutive CQ spectrograms will be overlapped too.

**Comment by leonid-s-usov** [comment]
So, if reconstruction is not my concern, should I pick the middle vector as representing the current frame?

**Comment by pabloEntropia** [comment]
That should work! You can also average along the time axis for a more robust estimation.
On the other hand, I will work on SpectrumCQ between today and tomorrow and I will notify you if I come up with a fix for it.

**Comment by leonid-s-usov** [comment]
@pabloEntropia thank you, really appreciate your support.
I'll update how NSGConstantQ works for me, in the meantime, may I ask if you have any information regarding the few other issues I've risen, whether I should expect responses there.
#887 #889 #891 #893

**Comment by dbogdanov** [comment]
Solved in PR #890.
